src/components/Surface/Surface.js

```diff
@@ -188,6 +188,26 @@ function Surface() {
           actionDropActive: false,
         };
       }
+      case 'setEditMode': {
+        const {
+          id,
+          editMode,
+        } = action.payload;
+        const newArguments = clone(state.argumentsList).map((argument) => {
```

Comment:
I don't think we need to clone the list before mapping?

```suggestion
        const newArguments = state.argumentsList.map((argument) => {
```